utilities/openemr-cmd/openemr-cmd

```diff
@@ -24,267 +30,292 @@ get_container_names() {
-		echo "e.g `basename $0` [docker-names|dn] php"
+		echo "e.g $(basename "$0") [docker-names|dn] php"
 		echo '***************************************************************'
 		echo '=======Running Docker Names======='
-		docker ps | awk '{print $NF}'|grep -v NAMES|sort -n
+		
+        # Capture output, check if empty
+        local running_containers
+        running_containers=$(docker ps --format "{{.Status}}\t{{.Names}}" | { grep -v NAMES || true; } | sort -n)
+        if [[ -z "${running_containers}" ]]; then
+            echo "No running containers found."
+        else
+		    echo '  STATUS        NAME'
+            echo "${running_containers}"
+        fi
 		echo '====Other Docker Status Names====='
-		docker ps -a --format "table{{.Status}}\t{{.Names}}"| grep -v Up
+        # Capture output, check if empty
+        local other_containers
+        other_containers=$(docker ps -a --format "{{.Status}}\t{{.Names}}" | grep -v Up || true)
+        if [[ -z "${other_containers}" ]]; then
+            echo "No containers found with other statuses."
+        else
+		    echo '  STATUS        NAME'
+            echo "${other_containers}"
+        fi
 	fi
 }
 
 restart_couchdb_docker(){
-  if [ -z "$COUCHDB_CONTAINER" ]
-  then
+  if [[ -z "${COUCHDB_CONTAINER}" ]]; then
     echo "Unable to find the couchdb docker, so could not restart it"
   else
     echo "Restarting couchdb docker"
-    docker restart "$COUCHDB_CONTAINER"
+    docker restart "${COUCHDB_CONTAINER}"
   fi
 }
 
 creat_a_backup_snapshot(){
-    BACKUP_FILE=$1
-    BACKUP_FILE_CODE=20
-    if [ $# != 1 ]
-    then
+    local BACKUP_FILE="$1"
+    local BACKUP_FILE_CODE=20
+    if [[ $# != 1 ]]; then
 		echo 'Please provide a snapshot name.'
 		echo 'e.g. openemr-cmd [backup-snapshot|bs] example'
-		exit $BACKUP_FILE_CODE
+		exit "${BACKUP_FILE_CODE}"
 	else
-		$DOCKER_EXEC_CMD $CONTAINER_ID $SHELL_TYPE "$DEV_TOOLS backup $BACKUP_FILE"
+		# Use the devtools function
+		run_devtools_in_docker "${CONTAINER_ID}" backup "${BACKUP_FILE}"
 	fi
 }
 
 restore_from_a_snapshot(){
-    BACKUP_FILE=$1
-    BACKUP_FILE_CODE=21
-    if [ $# != 1 ]
-    then
+    local BACKUP_FILE="$1"
+    local BACKUP_FILE_CODE=21
+    if [[ $# != 1 ]]; then
         echo 'Please provide a restore snapshot name.'
         echo 'e.g. openemr-cmd [restore-snapshot|rs] example'
-        exit $BACKUP_FILE_CODE
+        exit "${BACKUP_FILE_CODE}"
     else
-		$DOCKER_EXEC_CMD $CONTAINER_ID $SHELL_TYPE "$DEV_TOOLS restore $BACKUP_FILE"
+		# Use the devtools function
+		run_devtools_in_docker "${CONTAINER_ID}" restore "${BACKUP_FILE}"
 	fi
 }
 
 copy_capsule_from_docker_to_host(){
-	BACKUP_FILE=$1
-	BACKUP_HOST_DIR=$2 #optional parameter
-	BACKUP_FILE_CODE=19
-	if [ $# != 1 ] && [ $# != 2 ]
-	then
+	local BACKUP_FILE="$1"
+	local BACKUP_HOST_DIR="${2:-}" #optional parameter, default to empty
+	local BACKUP_FILE_CODE=19
+	if [[ $# != 1 ]] && [[ $# != 2 ]]; then
```

Comment:
```suggestion
	if (( $# != 1 && $# != 2 )); then
```

(Although if I had my druthers we'd flip negative if expressions and this'd start with `(( $# == 1 || $# == 2 ))`.)

Reply:
I just made a style change, not logic.